### PR TITLE
chore: improve logging on consensus failure

### PR DESCRIPTION
**Pull request by joschisan**
If such an error or panic occurs we need to read the pending accepted items of the current session from disk to obtain the item at which the consensus diverged.

### CODE REVIEW

File: fedimint-server/src/consensus/engine.rs [hunk 1]
```diff
@@ -393,13 +396,16 @@ impl ConsensusEngine {
                         if self.keychain.verify(&header, &signature, to_node_index(peer)){
                             signatures.insert(peer, signature);
                         } else {
-                            warn!(target: LOG_CONSENSUS, "Received invalid signature from peer {peer}");
+                            error!(target: LOG_CONSENSUS, "Consensus Failure: invalid header signature from {peer}");
```
Comment (elsirion) [inline fedimint-server/src/consensus/engine.rs:399]: Could we also print all our items here? I suppose all guardians would get that error and then we could compare?

File: fedimint-server/src/consensus/engine.rs [hunk 2]
```diff
@@ -350,7 +350,10 @@ impl ConsensusEngine {
                         .items
                         .split_at(pending_accepted_items.len());
 
-                    assert!(processed.iter().eq(pending_accepted_items.iter()));
+                    assert!(
+                        processed.iter().eq(pending_accepted_items.iter()),
+                        "Consensus Failure: pending accepted items disagree with federation consensus"
+                    );
```
Comment (elsirion) [inline fedimint-server/src/consensus/engine.rs:356]: If this was an `assert_eq` (over a `Vec`, idk about iterators) we would see the difference.

File: fedimint-server/src/consensus/engine.rs (hunk 1 above)
Comment (dpc) [reply fedimint-server/src/consensus/engine.rs:399]: Hmm... Actually, right...

If all guardians print all items whenever they receive a wrong signature, all prefixed if their ID, then it should be relatively easy to binary search the outstanding item.

We should probably print some very compact representation of all items. The exact item is not all that useful, but the module_ids on it are.

**Comment by dpc** [comment]
Follow-up: https://github.com/fedimint/fedimint/pull/5417

**Pull request merged**